python/tank/util/login.py

```diff
@@ -65,35 +68,47 @@ def get_shotgun_user(sg):
         g_shotgun_user_cache = sg.find_one("HumanUser", filters=[["login", "is", local_login]], fields=fields)
     
     return g_shotgun_user_cache
-        
+
 def get_current_user(tk):
     """
-    Retrieves the current user as a dictionary of metadata values.
-    Returns None if the user is not found in shotgun.
-    
-    Returns the following fields:
-    
-    * id
-    * type
-    * email
-    * login
-    * name
-    * image url (thumbnail)
-    
-    This method connects to shotgun.
-    """    
+    Retrieves the current user as a dictionary of metadata values. Note: This method connects to
+    shotgun the first time around. The result is then cached to reduce latency.
+    :returns: None if the user is not found in shotgun. Otherwise, it returns a dictionary
+              with the following fields:
+                 * id
+                 * type
+                 * email
+                 * login
+                 * name
+                 * image url (thumbnail)
+    """
     global g_shotgun_current_user_cache
-    if g_shotgun_current_user_cache == "unknown":
-    
+    if g_shotgun_current_user_cache != "unknown":
+        return g_shotgun_current_user_cache
+
+    from .. import api
+
+    user = api.get_current_user()
+
+    if sg_auth.is_script_user(user):
```

Comment:
I will probably comment on this further down (didn't get to that bit yet!) but I thought we already discussed having a single user object as part of the authentication API. Maybe it could be something like

```
    def get_user_entity():
        # returns sg entity for HumanUser or ScriptUser
        # returns link-dictionary of type, id and name
        #
        # {"type": "HumanUser", "id": 123, "name": "Manne Ohrstrom"}
        # {"type": "ApiUser", "id": 123, "name": "Toolkit"}
```

That feels a lot more straight forward to me and means we can drop the `is_session_user()` RTTI-style methods.